model/src/main/java/org/cbioportal/genome_nexus/model/ProteinLocation.java

```diff
@@ -0,0 +1,91 @@
+package org.cbioportal.genome_nexus.model;
+
+import java.util.*;
+
+public class ProteinLocation
+{
+    private String transcriptId;
+    private Integer start;
+    private Integer end;
+    private String mutationType;
+
+    public ProteinLocation() {}
+
+    public ProteinLocation(String transcriptId, Integer start, Integer end, String mutationType) {
+        this.transcriptId = transcriptId;
+        this.start = start;
+        this.end = end;
+        this.mutationType = mutationType;
+    }
+
+    public String getTranscriptId() {
+        return transcriptId;
+    }
+
+    public Integer getStart() {
+        return start;
+    }
+
+    public Integer getEnd() {
+        return end;
+    }
+
+    public String getMutationType() {
+        return mutationType;
+    }
+
+    public List<Hotspot> filterHotspot(List<Hotspot> hotspots, ProteinLocation proteinLocation){
+        int start = proteinLocation.getStart();
+        int end = proteinLocation.getEnd();
+        String type = proteinLocation.getMutationType();
+        List<Hotspot> result = new ArrayList<>();
+        for (Hotspot hotspot: hotspots) {
+            boolean valid = true;
+            
+            // Protein location
```

Comment:
We should have a separate method to parse `hotspotStart` and `hotspotEnd` from hotspot position. Or maybe you can use `Numerical.extractPositiveIntegers` method here?